docs/library/deflate.rst

```diff
@@ -134,44 +139,43 @@ Deflate window size
-mean that the stream will require the same size window for decompression, as
-the stream may not reference data as far back as the window allows (for example,
-if the length of the input is smaller than the window size).
-
-If the decompressor uses a smaller window size than necessary for the input data
-stream, it will fail mid-way through decompression with :exc:`OSError`.
-
-.. _deflate_wbits_zlib:
-
-The zlib format includes a header which specifies the window size used to
-compress the data (which due to the above, may be larger than the size required
-for the decompressor).
-
-If this header value is lower than the specified *wbits* value, then the header
-value will be used instead in order to reduce the memory allocation size. If
-the *wbits* parameter is zero (the default), then the header value will only be
-used if it is less than the maximum value of ``15`` (which is default value
-used by most compressors [#f1]_).
-
-In other words, if the source zlib stream has been compressed with a custom window
-size (i.e. less than ``15``), then using the default *wbits* parameter of zero
-will decompress any such stream.
-
-The gzip file format does not include the window size in the header.
-Additionally, most compressor libraries (including CPython's implementation
-of :class:`gzip.GzipFile`) will default to the maximum possible window size.
-This makes it difficult to decompress most gzip streams on MicroPython unless
-your board has a lot of free RAM.
-
-If you control the source of the compressed data, then prefer to use the zlib
-format, with a window size that is suitable for your target device.
-
-.. rubric:: Footnotes
-
-.. [#f1] The assumption here is that if the header value is the default used by
-   most compressors, then nothing is known about the likely required window
-   size and we should ignore it.
+reference. Increasing the window size will improve compression, but will require
+more memory and make the compressor slower.
+
+If an input stream was compressed a given window size, then `DeflateIO`
+using a smaller window size will fail mid-way during decompression with
+:exc:`OSError`, but only if a back-reference actually refers back further
+than the decompressor's window size. This means it may be possible to decompress
+with a smaller window size. For example, this would trivially be the case if the
+original uncompressed data is shorter than the window size.
+
+Decompression
+~~~~~~~~~~~~~
+
+The zlib format includes a header which specifies the window size that was used
+to compress the data. This indicates the maximum window size required to
+decompress this stream. If this header value is less than the specified *wbits*
+value (or if *wbits* is unset), then the header value will be used.
+
+The gzip format does not include the window size in the header, and assumes that
+all gzip compressors (e.g. the `gzip` utility, or CPython's implementation of
```

Comment:
WARNING: more than one target found for 'any' cross-reference 'gzip': could be :doc:`gzip – gzip compression & decompression` or :py:mod:`gzip: gzip compression & decompression`

Reply:
Fixed, thanks.